sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -408,7 +411,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
    *
    * @param <T> the type of values written to the sink.
    */
-  public abstract static class FileBasedWriteOperation<T> implements Serializable {
+  public abstract static class WriteOperation<T> implements Serializable {
```

Comment:
I'm trying to be more of a stickler about clean history, for code spelunking purposes. It would be very helpful to rebase your commit on top of one that is limited to _just_ this rename as an automated refactor (plus any needed style fixups). Happy to help with that or clarify what I mean.